App/Views/SearchView.swift

```diff
@@ -0,0 +1,1238 @@
+                navAppearance.titleTextAttributes = [.foregroundColor: textColor]
+                
+                navigationController.navigationBar.standardAppearance = navAppearance
+                navigationController.navigationBar.scrollEdgeAppearance = navAppearance
+                
+                // Configure toolbar
+                let toolbarAppearance = UIToolbarAppearance()
+                toolbarAppearance.configureWithOpaqueBackground()
+                toolbarAppearance.backgroundColor = theme[uicolor: "tabBarBackgroundColor"]
+                toolbarAppearance.shadowColor = .clear
+                
+                navigationController.toolbar.standardAppearance = toolbarAppearance
+                navigationController.toolbar.compactAppearance = toolbarAppearance
+                if #available(iOS 15.0, *) {
+                    navigationController.toolbar.scrollEdgeAppearance = toolbarAppearance
+                    navigationController.toolbar.compactScrollEdgeAppearance = toolbarAppearance
+                }
+                
+                // Force immediate update
+                navigationController.toolbar.setNeedsLayout()
+            }
+        }
+    }
+}
+
+// MARK: - Preview Provider
+struct SearchView_Previews: PreviewProvider {
+    static var previews: some View {
+        SearchView(previewModel: SearchPageViewModel(
+            forumSelectOptions: [
+                ForumSelectOption(optionText: "Main Forum", value: "1"),
+                ForumSelectOption(optionText: "Sub Forum", value: "2", depth: 1, parentIDs: ["1"]),
+                ForumSelectOption(optionText: "Another Forum", value: "3")
+            ],
+            searchHelpHints: [
+                SearchHelpHint(text: "Search by title: intitle:\"example\""),
+                SearchHelpHint(text: "Search by user: username:\"example\"")
+            ]
+        ))
+        .environment(\.theme, Theme.theme(named: "brightLight") ?? Theme.defaultTheme())
+    }
+}
+
+// thanks to https://swiftwithmajid.com/2020/03/04/customizing-toggle-in-swiftui/
+struct CheckboxToggleStyle: ToggleStyle {
+    @SwiftUI.Environment(\.isEnabled) var isEnabled
+    @SwiftUI.Environment(\.theme) var theme
+    
+    func makeBody(configuration: Configuration) -> some View {
+        HStack {
+            configuration.label
+            Spacer()
+            Image(systemName: configuration.isOn ? "checkmark.square.fill" : "square")
+                .resizable()
+                .frame(width: 22, height: 22)
+                .foregroundColor(configuration.isOn ? theme[color: "tintColor"]! : isEnabled ? theme[color: "listTextColor"]! : theme[color: "placeholderTextColor"]!)
+        }
+        .font(.body)
+        .contentShape(Rectangle())
+        .onTapGesture {
+            withAnimation {
+                configuration.isOn.toggle()
+            }
+        }
+    }
+}
+
+// MARK: - Constants
+private let defaultSearchPageHTMLString =  """
```

Comment:
why is this hardcoded? we should be scraping it so the app adapts whenever it changes on the forums.

Reply:
This was hardcoded with the idea that it would get overwritten by the site data. But apparently that second overwriting part was on the TODO list and forgotten.
I've removed it and we use actual website data now.

The purpose was mostly for previews. I tried to use fixtures, but it didn't work out. I came up with a simpler preview solution tho.

Reply:
all good! ty